fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/SingleLoanChargeRepaymentScheduleProcessingWrapper.java

```diff
@@ -46,7 +47,8 @@ public void reprocess(final MonetaryCurrency currency, final LocalDate disbursem
             totalPrincipal = totalPrincipal.plus(installment.getPrincipal(currency));
         }
         LoanChargePaidBy accrualBy = null;
-        if (!loan.isInterestBearing() && loanCharge.isSpecifiedDueDate()) { // TODO: why only if not interest bearing
+        if ((!loan.isInterestBearing() || loan.getLoanProductRelatedDetail().getLoanScheduleType().equals(LoanScheduleType.PROGRESSIVE))
```

Comment:
Would you mind to remove the `!loan.isInterestBearing()` as well?
We should only have this:
`if (loanCharge.isSpecifiedDueDate())`,